.github/workflows/release.yml

```diff
@@ -15,54 +15,21 @@ name: Release
 jobs:
   build:
     uses: ./.github/workflows/continuous-integration.yml
-  upload:
-    name: Upload Release Asset
-    needs: build
-    runs-on: windows-latest
 
-    steps:
-
-      # Using shared artifact from build workflow
-      - name: Download Artifact
-        uses: actions/download-artifact@c850b930e6ba138125429b7e5c93fc707a7f8427 # v4.1.4
-        with:
-          name: shell-binaries
-          path: ${{ github.workspace }}/shell-binaries
-    
-      - name: Display structure of downloaded files
-        run: Get-ChildItem -Recurse
-        working-directory: ${{ github.workspace }}/shell-binaries
-        
-      - name: Compress
-        run: Compress-Archive -Path  ${{ github.workspace }}/shell-binaries/* composeui-${{ github.ref_name }}-win32.zip
-        
-      - name: Upload Release Asset
-        id: upload-release-asset 
-        uses: actions/upload-release-asset@e8f9f06c4b078e705bd2ea027f0926603fc9b4d5 # v1.0.2
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ github.event.release.upload_url }}
-          asset_path: ./composeui-${{ github.ref_name }}-win32.zip
-          asset_name: composeui-${{ github.ref_name }}-win32.zip
-          asset_content_type: application/zip
-          
-  deploy-npm:
-    name: Publish Package to npmjs
+  deploy_to_nuget:
+    name: Publish messaging packages to nuget.org
     runs-on: windows-latest
-    needs: upload
+    needs: build
     steps:
-      - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
-      - uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # v4.0.2
+      - uses: actions/download-artifact@c850b930e6ba138125429b7e5c93fc707a7f8427 # v4.1.4
         with:
-          node-version: 18
-          registry-url: https://registry.npmjs.org/
-      - run: |
+        name: packages
+        path: ./packages
+      - uses: actions/setup-dotnet@4d6c8fcf3c8f7a60068d26b594648e99df24cee3 # v4.0.0
+        with:
+          dotnet-version: ${{ matrix.dotnet-version }}
```

Comment:
I don't see a matrix defined in here but also, why a matrix, wouldn't we be overwriting each package?

Reply:
Ideally I'd like to have a central place to define what versions are we using instead of defining .net (and node) versions all over the place in order to reduce the cognitive load and the chance for human error. Using a matrix seemed to be the closest I can get to this without ugly hacks, and I saw Maria using it before. I think I got lost in the scopes of things though so let me revisit this idea :)

Reply:
Sidenote: I think I found a way to achieve this without hacking, but that's out of scope for this, so I just hardcoded the .net version for now

Reply:
I think you would move the setup-dotnet job into its own yml and have it as a required of the other actions